src/store/modules/DownloadModule.ts

```diff
@@ -211,6 +214,30 @@ export const DownloadModule = {
                 state.allDownloads = newDownloads;
             }
         },
+        setDone(state: State, downloadId: number) {
+            const index: number = getIndexOfDownloadProgress(state.allDownloads, downloadId);
```

Comment:
Could there be another helper function so this would become something like
`state.allDownloads = updateDownloadStatus(state.allDownloads, downloadId, DownloadStatus.DONE);`
and then use it in the other commits too to avoid the repetition?